### PR TITLE
🐛Source Airtable: wrap HTTPError 400 in AirbyteTracedException

### PR DESCRIPTION
**Pull request by askarpets**
## What
Avoid raising `HTTPError` with code 400 on getting `refresh_token` operation

## How
Raise `AirbyteTracedException` in case `refresh_token` response returned with status code of 400 and error type of `invalid_grant`

## Recommended reading order
1. `source_airtable/auth.py`

## 🚨 User Impact 🚨
No breaking changes

## Pre-merge Actions
<details><summary><strong>Updating a connector</strong></summary>

### Community member or Airbyter

- Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- Unit & integration tests added

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- Create a non-forked branch based on this PR and test the below items on it
- Build is successful
- If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).

</details>

### CODE REVIEW

**Comment by airbyte-oss-build-runner** [comment]
##  source-airtable test report (commit [`1f042fb89b`](https://github.com/airbytehq/airbyte/pull/31119/commits/1f042fb89b78460f126ef103f5ab36b3ec15c393)) - ❌

⏲️  Total pipeline duration: 02mn04s 

| Step                                                            | Result   |
|:----------------------------------------------------------------|:---------|
| Connector package install                                       | ✅        |
| Build source-airtable docker image for platform(s) linux/x86_64 | ✅        |
| Unit tests                                                      | ✅        |
| Acceptance tests                                                | ❌        |
| Code format checks                                              | ✅        |
| Validate metadata for source-airtable                           | ✅        |
| Connector version semver check                                  | ✅        |
| Connector version increment check                               | ✅        |
| QA checks                                                       | ❌        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/25837-wrap-http-error/1696580703/1f042fb89b78460f126ef103f5ab36b3ec15c393/source-airtable/3.0.2/output.html)

☁️ [View runs for commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:1f042fb89b78460f126ef103f5ab36b3ec15c393)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connectors/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-airtable test
```

**Comment by airbyte-oss-build-runner** [comment]
##  source-airtable test report (commit [`1f042fb89b`](https://github.com/airbytehq/airbyte/pull/31119/commits/1f042fb89b78460f126ef103f5ab36b3ec15c393)) - ❌

⏲️  Total pipeline duration: 01mn11s 

| Step                                                            | Result   |
|:----------------------------------------------------------------|:---------|
| Connector package install                                       | ✅        |
| Build source-airtable docker image for platform(s) linux/x86_64 | ✅        |
| Unit tests                                                      | ✅        |
| Acceptance tests                                                | ❌        |
| Code format checks                                              | ✅        |
| Validate metadata for source-airtable                           | ✅        |
| Connector version semver check                                  | ✅        |
| Connector version increment check                               | ✅        |
| QA checks                                                       | ❌        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/25837-wrap-http-error/1696590518/1f042fb89b78460f126ef103f5ab36b3ec15c393/source-airtable/3.0.2/output.html)

☁️ [View runs for commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:1f042fb89b78460f126ef103f5ab36b3ec15c393)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connectors/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-airtable test
```

File: airbyte-integrations/connectors/source-airtable/source_airtable/auth.py [hunk 1]
```diff
@@ -42,8 +44,15 @@ def _get_refresh_access_token_response(self) -> Mapping[str, Any]:
             data=self.build_refresh_request_body(),
             headers=self.build_refresh_request_headers(),
         )
+        content = response.json()
+        if response.status_code == 400 and content.get("error") == "invalid_grant":
+            raise AirbyteTracedException(
+                internal_message=content.get("error_description"),
+                message="Refresh token is invalid. Please update it in the config.",
```
Comment (darynaishchenko) [inline airbyte-integrations/connectors/source-airtable/source_airtable/auth.py:51]: We need to provide understandable user messages, users click 'Authenticate' button in set up page, which doesn't show refresh token field. if we want to ask user create a new one we should ask him to 'Re-authenticate'. 

`Refresh token is invalid/expired. Please re-authenticate to restore access to Airtable.` - more suitable error message.

**Review by darynaishchenko — CHANGES_REQUESTED** [review]

**Review by roman-yermilov-gl — CHANGES_REQUESTED** [review]
If version is increased then [documentation](https://github.com/airbytehq/airbyte/blob/master/docs/integrations/sources/airtable.md) should also be updated. Please update changelog section with your version/PR link/description info

File: airbyte-integrations/connectors/source-airtable/source_airtable/auth.py (hunk 1 above)
Comment (askarpets) [reply airbyte-integrations/connectors/source-airtable/source_airtable/auth.py]: Thanks for the advice, updated

**Comment by airbyte-oss-build-runner** [comment]
##  source-airtable test report (commit [`c2dd9c5e6b`](https://github.com/airbytehq/airbyte/pull/31119/commits/c2dd9c5e6b999008a271d881a50ca1b25702f7b4)) - ❌

⏲️  Total pipeline duration: 02mn01s 

| Step                                                            | Result   |
|:----------------------------------------------------------------|:---------|
| Connector package install                                       | ✅        |
| Build source-airtable docker image for platform(s) linux/x86_64 | ✅        |
| Unit tests                                                      | ✅        |
| Acceptance tests                                                | ❌        |
| Code format checks                                              | ✅        |
| Validate metadata for source-airtable                           | ✅        |
| Connector version semver check                                  | ✅        |
| Connector version increment check                               | ✅        |
| QA checks                                                       | ✅        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/25837-wrap-http-error/1696591838/c2dd9c5e6b999008a271d881a50ca1b25702f7b4/source-airtable/3.0.2/output.html)

☁️ [View runs for commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:c2dd9c5e6b999008a271d881a50ca1b25702f7b4)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connectors/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-airtable test
```

**Review by darynaishchenko — APPROVED** [review]

**Review by roman-yermilov-gl — APPROVED** [review]

**Review by bnchrch — APPROVED** [review]

**Comment by airbyte-oss-build-runner** [comment]
##  source-airtable test report (commit [`df9b9c792a`](https://github.com/airbytehq/airbyte/pull/31119/commits/df9b9c792ab32ce0c3a01754379d95df69321046)) - ❌

⏲️  Total pipeline duration: 02mn24s 

| Step                                                            | Result   |
|:----------------------------------------------------------------|:---------|
| Connector package install                                       | ✅        |
| Build source-airtable docker image for platform(s) linux/x86_64 | ✅        |
| Unit tests                                                      | ✅        |
| Acceptance tests                                                | ❌        |
| Code format checks                                              | ✅        |
| Validate metadata for source-airtable                           | ✅        |
| Connector version semver check                                  | ✅        |
| Connector version increment check                               | ✅        |
| QA checks                                                       | ✅        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/25837-wrap-http-error/1696937003/df9b9c792ab32ce0c3a01754379d95df69321046/source-airtable/4.1.1/output.html)

☁️ [View runs for commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:df9b9c792ab32ce0c3a01754379d95df69321046)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connectors/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-airtable test
```

**Comment by airbyte-oss-build-runner** [comment]
##  source-airtable test report (commit [`df9b9c792a`](https://github.com/airbytehq/airbyte/pull/31119/commits/df9b9c792ab32ce0c3a01754379d95df69321046)) - ❌

⏲️  Total pipeline duration: 02mn08s 

| Step                                                            | Result   |
|:----------------------------------------------------------------|:---------|
| Connector package install                                       | ✅        |
| Build source-airtable docker image for platform(s) linux/x86_64 | ✅        |
| Unit tests                                                      | ✅        |
| Acceptance tests                                                | ❌        |
| Code format checks                                              | ✅        |
| Validate metadata for source-airtable                           | ✅        |
| Connector version semver check                                  | ✅        |
| Connector version increment check                               | ✅        |
| QA checks                                                       | ✅        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/25837-wrap-http-error/1696937751/df9b9c792ab32ce0c3a01754379d95df69321046/source-airtable/4.1.1/output.html)

☁️ [View runs for commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:df9b9c792ab32ce0c3a01754379d95df69321046)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connectors/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-airtable test
```

**Comment by airbyte-oss-build-runner** [comment]
##  source-airtable test report (commit [`e70d4bd63e`](https://github.com/airbytehq/airbyte/pull/31119/commits/e70d4bd63e63a07a608c994a947b0c6124b8a1b7)) - ❌

⏲️  Total pipeline duration: 01mn08s 

| Step                                                            | Result   |
|:----------------------------------------------------------------|:---------|
| Connector package install                                       | ✅        |
| Build source-airtable docker image for platform(s) linux/x86_64 | ✅        |
| Unit tests                                                      | ✅        |
| Acceptance tests                                                | ❌        |
| Code format checks                                              | ✅        |
| Validate metadata for source-airtable                           | ✅        |
| Connector version semver check                                  | ✅        |
| Connector version increment check                               | ✅        |
| QA checks                                                       | ✅        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/25837-wrap-http-error/1696939203/e70d4bd63e63a07a608c994a947b0c6124b8a1b7/source-airtable/4.1.1/output.html)

☁️ [View runs for commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:e70d4bd63e63a07a608c994a947b0c6124b8a1b7)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connectors/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-airtable test
```

**Comment by airbyte-oss-build-runner** [comment]
##  source-airtable test report (commit [`e70d4bd63e`](https://github.com/airbytehq/airbyte/pull/31119/commits/e70d4bd63e63a07a608c994a947b0c6124b8a1b7)) - ❌

⏲️  Total pipeline duration: 02mn05s 

| Step                                                            | Result   |
|:----------------------------------------------------------------|:---------|
| Connector package install                                       | ✅        |
| Build source-airtable docker image for platform(s) linux/x86_64 | ✅        |
| Unit tests                                                      | ✅        |
| Acceptance tests                                                | ❌        |
| Code format checks                                              | ✅        |
| Validate metadata for source-airtable                           | ✅        |
| Connector version semver check                                  | ✅        |
| Connector version increment check                               | ✅        |
| QA checks                                                       | ✅        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/25837-wrap-http-error/1696939670/e70d4bd63e63a07a608c994a947b0c6124b8a1b7/source-airtable/4.1.1/output.html)

☁️ [View runs for commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:e70d4bd63e63a07a608c994a947b0c6124b8a1b7)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connectors/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-airtable test
```

**Comment by airbyte-oss-build-runner** [comment]
##  source-airtable test report (commit [`068d81a7eb`](https://github.com/airbytehq/airbyte/pull/31119/commits/068d81a7ebfb6337dc9d3b096294693a1a6ea6b8)) - ✅

⏲️  Total pipeline duration: 01mn09s 

| Step                                                            | Result   |
|:----------------------------------------------------------------|:---------|
| Connector package install                                       | ✅        |
| Build source-airtable docker image for platform(s) linux/x86_64 | ✅        |
| Unit tests                                                      | ✅        |
| Acceptance tests                                                | ✅        |
| Code format checks                                              | ✅        |
| Validate metadata for source-airtable                           | ✅        |
| Connector version semver check                                  | ✅        |
| Connector version increment check                               | ✅        |
| QA checks                                                       | ✅        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/25837-wrap-http-error/1696947672/068d81a7ebfb6337dc9d3b096294693a1a6ea6b8/source-airtable/4.1.1/output.html)

☁️ [View runs for commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:068d81a7ebfb6337dc9d3b096294693a1a6ea6b8)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connectors/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-airtable test
```

**Pull request merged**